apertium-tam.tam.rlx

```diff
@@ -1,15 +1,77 @@
-# Firstly, we need to define what tags should be considered sentence delimiters. For this example, only full stop is set as delimiter.
-DELIMITERS = "<.>" ;
+DELIMITERS = "<.>" "<!>" "<?>" "<...>" ;
 
-# We can define sets for common tag clusters
-LIST Case = (n s) (n p) ; # A set that matches either a Noun Singular or a Noun Plural
+SETS
+
+LIST N = n;
+LIST Adj = adj;
+LIST Adv = adv;
+LIST Verb = vblex;
+LIST Vbser = vbser;
+LIST Prn = prn;
+LIST MaleNoun = (n m);
+LIST FemaleNoun = (n f);
+
+LIST Sg = sg;
+LIST Pl = pl;
+LIST P1 = p1;
+LIST P2 = p2;
+LIST P3 = p3;
+LIST M = m;
+LIST F = f;
 
 SECTION
-# If there is a singular noun to the right, I cannot be a verb or noun.
-REMOVE (n) OR (v) IF (1 (n s)) ;
 
-# If there is a conjunction followed by a certain cohort of the same CASE as me, choose me.
-SELECT $$Case IF (1 (cnjcoo) LINK 1C $$Case) ;
+# NOUN
+
+# Select noun if preceded by a adjective.
+SELECT N IF (0 N) (0 V) (-1 Adj) ;
+
+
+# Adjective
+
+# Select adjective if followed by a noun.
+SELECT Adj IF (0 Adj) (0 Adv) (1 N);
+
+
+# Adverb
+
+# Select adverb if followed by a noun.
+SELECT Adv IF (0 Adj) (0 Adv) (1 Verb);
+
+
+# VERBS
+
+# Select noun if preceded by a adjective.
+SELECT V IF (0 N) (0 V) (-1 Adv) ;
+
+# Prefer singular for verbs if preceded by a singular noun or pronoun
+SELECT Sg IF (o Verb) (-1* (n sg));
+SELECT Sg IF (o Verb) (-1* (prn sg));
+
+# Prefer plural for verbs if preceded by a plural noun or pronoun
+SELECT Pl IF (o Verb) (-1* (n pl));
+SELECT Pl IF (o Verb) (-1* (prn pl));
+
+# Prefer 1st person for verbs if preceded by a 1st person noun or pronoun
+SELECT P1 IF (o Verb) (-1* (n p1));
+SELECT P1 IF (o Verb) (-1* (prn p1));
+
+# Prefer 2nd person for verbs if preceded by a 2nd person noun or pronoun
+SELECT P2 IF (o Verb) (-1* (n p2));
+SELECT P2 IF (o Verb) (-1* (prn p2));
+
+# Prefer 3rd person for verbs if preceded by a 3rd person noun or pronoun
+SELECT P3 IF (o Verb) (-1* (n p3));
+SELECT P3 IF (o Verb) (-1* (prn p3));
+
+# Prefer masculine for verbs if preceded by a masculine noun or pronoun
+SELECT M IF (o Verb) (-1* (n m));
+SELECT M IF (o Verb) (-1* (prn m));
+
+# Prefer feminine for verbs if preceded by a feminine noun or pronoun
+SELECT F IF (o Verb) (-1* (n f));
+SELECT F IF (o Verb) (-1* (prn f));
```

Comment:
These should also have `0` rather than `o`. Also, maybe it would be cleaner to create a "noun or pronoun" set so you don't have to duplicate all the rules.

Reply:
Okay. :)